### ISSUE TITLE
Fails silently on valid RAML, even with names option

### ISSUE THREAD

**Issue by malowitzki** [cybertk/abao#230]
Using abao to test my REST API, which is running in the default Docker machine:

```me$ abao my-api.raml --server $(docker-machine ip)```

Hangs for <.5s, then abao exits and I'm back at my prompt, nothing output or printed by abao. Even when I run with just the names option (`me$ abao my-api.raml --names`), it exits with no output. If it's failing or throwing an error, it's doing so silently; so I have no good way to troubleshoot.

Initially, some of my JSON schema were invalid, and abao was (correctly) complaining; so I know abao is working to that degree, and I have now confirmed that all my JSON and RAML is valid.

**Comment by plroebuck** [comment]
Dunno your Dockerfile setup and it's not provided. Ensure your outside:inside port#s are in agreement. The `server` argument should be a URL in the form 'http://host-or-ip-addr:outside-port'. Most  likely scenario is that **abao** is working but `server` argument is invalid. Verify this by using **curl** with the same `server` argument to verify your REST endpoint is externally accessible.

**Comment by malowitzki** [comment]
I've successfully hit several endpoints in the Dockerized app, both through curl and Postman. In both curl and Postman, the URL I use is just the IP address plus the path (e.g. '192.168.99.100/my-api/v1/...'). I tried adding the 'http://' scheme to the front of it, and the outside port ':8000' (which I believe is the default) when using abao, but still nothing.

IMO, the more troubling issue is that abao is failing _silently_. Obviously I want to get it working for myself, but to exit without any further info is frustrating for any user.

**Comment by plroebuck** [comment]
Again, without seeing your Dockerfile, this is all guesswork. If you're using "192.168.99.100/my-api/v1/..." with **curl** and getting output, then your REST server is probably Apache-fronted (redirecting from 80 to 8000). Correspondingly, your `server` argument should be "http://192.168.99.100" (i.e., default to port 80). Have you verified your requests are making it into the Docker machine? Check Apache logfile to verify request recieved, and your REST server logfile to see actual request.



Based on what you stated, your `server` argument to **abao** should be 'http://docker-machine-ip:8000'

**Comment by malowitzki** [comment]
@plroebuck Thanks for rephrasing, that wording was a little clearer for me to follow. 

I have checked the Dockerfile and confirmed that the app is running on the default port (80), and also that my requests from curl and Postman are being logged. I've tried every combination of scheme, IP address, and port number in the server argument. Still nothing 👎

**Comment by plroebuck** [comment]
How about attach your Dockerfile and simplified RAML (single resource with GET for debugging)?

How is server being run? What language/framework was it written in? Would not usually expect REST endpoint to assume control of port 80; usually on 8000 or 8080 instead as ports below 1024 reserved for system-level services.

**Comment by malowitzki** [comment]
Ok, I have abao working with this (most basic ever) Hello world! app. I tried to add as many layers of complexity (e.g. docker-compose, a very prescriptive schema) without moving too far past Hello world!

I'm working to rule out some factors with the much more complex app that I want to test—but this here is working.

[abao-test.zip](https://github.com/cybertk/abao/files/1356364/abao-test.zip)

**Comment by plroebuck** [comment]
Given:
``$ SERVICE="abao-test"``
``$ DOCKER_MACHINE="192.168.99.100"``
``$ PORT="3000"``
``$ HOSTPORT="${DOCKER_MACHINE}:${PORT}"``
``$ RAML_FILE="${SERVICE}.raml"``

Assume that running test app is accessible via **curl** using:

``$ curl http://${HOSTPORT}/${SERVICE}/v1/``

and accessible via **abao** using:

``$ abao ${RAML_FILE} --server http://${HOSTPORT}``

**Comment by malowitzki** [comment]
Yes, everything in the abao-test example is working just fine. I did that exercise to confirm that abao will validate even the most basic example of an app running in Docker on my machine, and indeed it works. However, it still fails silently when I try it on my much larger Dockerized app, which I can hit no problem from both Postman and curl.

My latest idea was to change the port mapping in my docker-compose.yml so that I have an explicit mapping from 8000 to 80. I thought maybe abao was having some issue hitting 80. I confirmed that the new mapping works in Postman (by hitting http://192.168.99.100:8000/my-api/v1/...), but when I try `abao my-api.raml --server http://192.168.99.100:8000`, I still get the same result: silent exit.

**Comment by plroebuck** [comment]
I was trying to verify my understanding of your setup with my previous post; you didn't exactly say that those exact commands work with test app.

Assuming your original was also NodeJS service, you did verify that internal port from "docker-compose.yml" matches the listening port in "app.js", right? I would avoid using 80 as external port myself regardless.

**Comment by malowitzki** [comment]
The original service is a Java Spring Boot application. It's running a load balancer, so that's been pitched as a possible complication—But as far as curl and Postman are concerned, the app is working as expected.

I'm sorry I can't be more helpful. If abao was logging anything at all before exiting, I could give you something to work off of; but it's giving me nothing. (tbh, I think this is an opportunity to improve abao; and I would contribute by adding in the missing error logging if I had any inkling as to where it's failing.)
